src/Nethermind/Nethermind.Trie/TrieNode.cs

```diff
@@ -1340,5 +1345,151 @@ static void ThrowNotPersisted()
                 throw new InvalidOperationException("Cannot unresolve a child that is not persisted yet.");
             }
         }
+
+        public ChildIterator CreateChildIterator()
+        {
+            return new ChildIterator(this);
+        }
+
+        // Allow faster forward child iteration by not re-skipping items on each child seek
+        public ref struct ChildIterator(TrieNode node)
+        {
+            private ValueRlpStream _rlpStream;
+            private int? _currentStreamIndex;
+
+            private object? ResolveChildWithChildPath(ITrieNodeResolver tree, ref TreePath childPath, int i)
+            {
+                object? childOrRef;
+                SpanSource rlp = node._rlp;
+                ref var data = ref node._nodeData[i];
+                if (rlp.IsNull)
+                {
+                    childOrRef = data;
+                }
+                else
+                {
+                    if (data is null)
+                    {
+                        if (_currentStreamIndex.HasValue && _currentStreamIndex <= i)
+                        {
+                            int toSkip = i - _currentStreamIndex.Value;
+                            for (int j = 0; j < toSkip; j++) _rlpStream.SkipItem();
+                            _currentStreamIndex += toSkip;
+                        }
+                        else
+                        {
+                            _rlpStream = new ValueRlpStream(rlp);
+                            _rlpStream.Reset();
+                            _rlpStream.SkipLength();
+                            if (node.IsExtension)
+                            {
+                                _rlpStream.SkipItem();
+                                i--;
+                            }
+                            else
+                            {
+                                for (int j = 0; j < i; j++) _rlpStream.SkipItem();
+                            }
+
+                            _currentStreamIndex = i;
+                        }
+
+                        int prefix = _rlpStream.ReadByte();
+
+                        switch (prefix)
+                        {
+                            case 0:
+                            case 128:
```

Comment:
This is outside the scope of this PR (and not introduced here), but is there any reason to use magic numbers for RLP instead of consts?